### PR TITLE
[WIP] drop TypeConfig and type reference

### PR DESCRIPTION
**Pull request by giovannialbero1992**
In ref. to [#1569](https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1569)

### CODE REVIEW

**Comment by giovannialbero1992** [comment]
@deguif the tests are green. But I'm not convinced by FosElasticaExtension. I'd like to drop reference to type concept like loadTypeConfig and also into RepositoryManager. What do you suggest?

File: src/DependencyInjection/FOSElasticaExtension.php [hunk 1]
```diff
@@ -502,35 +497,24 @@ private function loadObjectPersister(array $typeConfig, Reference $indexRef, Con
     /**
      * Loads a pager provider for a type.
```
Comment (leobeal) [inline src/DependencyInjection/FOSElasticaExtension.php]: Should be `Loads a pager provider for an index`

File: src/Configuration/IndexConfig.php [hunk 2]
```diff
@@ -35,17 +35,20 @@ class IndexConfig implements IndexConfigInterface
     /**
      * Constructor expects an array as generated by the Container Configuration builder.
      *
-     * @param string       $name
-     * @param TypeConfig[] $types
-     * @param array        $config
+     * @param string $name
```
Comment (leobeal) [inline src/Configuration/IndexConfig.php]: docblocks are incorrect

**Comment by giovannialbero1992** [comment]
@deguif can you merge this PR and update #1569? @leobeal is using this fork and all seems to work.

File: .phpunit.result.cache [hunk 3]
```diff
@@ -0,0 +1 @@
...er\ModelToElasticaAutoTransformerTest::testObjectsMappingOfAtLeastOneAutoMappedObjectAndAtLeastOneManuallyMappedObject";d:0;s:124:"FOS\ElasticaBundle\Tests\Unit\Transformer\ModelToElasticaAutoTransformerTest::testThatMappedObjectsDontNeedAnIdentifierField";d:0;s:124:"FOS\ElasticaBundle\Tests\Unit\Transformer\ModelToElasticaAutoTransformerTest::testThatNestedObjectsDontNeedAnIdentifierField";d:0;s:117:"FOS\ElasticaBundle\Tests\Unit\Transformer\ModelToElasticaAutoTransformerTest::testNestedTransformHandlesSingleObjects";d:0;s:129:"FOS\ElasticaBundle\Tests\Unit\Transformer\ModelToElasticaAutoTransformerTest::testNestedTransformReturnsAnEmptyArrayForNullValues";d:0;s:123:"FOS\ElasticaBundle\Tests\Unit\Transformer\ModelToElasticaAutoTransformerTest::testUnmappedFieldValuesAreNormalisedToStrings";d:0;s:108:"FOS\ElasticaBundle\Tests\Unit\Transformer\ModelToElasticaAutoTransformerTest::testIdentifierIsCastedToString";d:0;s:117:"FOS\ElasticaBundle\Tests\Unit\Transformer\ModelToElasticaIdentifierTransformerTest::testGetDocumentWithIdentifierOnly";d:0;s:127:"FOS\ElasticaBundle\Tests\Unit\Transformer\ModelToElasticaIdentifierTransformerTest::testGetDocumentWithIdentifierOnlyWithFields";d:0;}}}
```
Comment (deguif) [inline .phpunit.result.cache]: Should be reverted.

File: src/Configuration/IndexTemplateConfig.php [hunk 4]
```diff
@@ -9,32 +9,35 @@
...
-     * Index name pattern
-     *
-     * @var string
-     */
-    private $template;
-
     /**
      * Constructor expects an array as generated by the Container Configuration builder.
+     *
+     * @param array $config
```
Comment (deguif) [inline src/Configuration/IndexTemplateConfig.php]: The type-hint doesn't bring any value here, I think it can be removed.

**Review by deguif — APPROVED** [review]

**Comment by deguif** [comment]
@giovannialbero1992 just minor comments.
Looks good to me, but just have a quick look.
Once merged on the original PR, I will update my code base patch in order to deeper test it ;)

File: src/DependencyInjection/FOSElasticaExtension.php [hunk 5]
```diff
@@ -291,43 +291,38 @@ private function loadIndexFinder(ContainerBuilder $container, string $name, Refe
...
+            'analyzer',
+            'search_analyzer',
+            'dynamic',
+            'date_detection',
+            'dynamic_date_formats',
+            'numeric_detection',
+        ] as $field) {
+            $indexConfig['config'][$field] = array_key_exists($field, $index) ?
+                $index[$field] :
+                null;
```
Comment (deguif) [inline src/DependencyInjection/FOSElasticaExtension.php]: This indentations seems weird to me, but maybe that's just me.

**Comment by deguif** [comment]
Nice let's merge and see it in action ;)

**Pull request merged**